### ISSUE TITLE
Custom replacer and reviver for JSON transforms?

### ISSUE THREAD

**Issue by RanzQ** [michaelcontento/redux-storage-engine-localstorage#19]
I would change the api to take parameters `key, replacer, reviver`. That way it would be easy to control how to store special objects, like moment.js in my case. I've used this with `redux-localstorage`:

```
import moment from 'moment'
const RE_ISO_DATE = /(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z))|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d([+-][0-2]\d:[0-5]\d|Z))|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d([+-][0-2]\d:[0-5]\d|Z))/

export function replacer (k, v) {
  if (typeof v === 'string' && v.match(RE_ISO_DATE)) {
    return 'moment:' + moment(v).valueOf()
  }
  return v
}

export function reviver (k, v) {
  if (typeof v === 'string' && v.includes('moment:')) {
    return moment(parseInt(v.split(':')[1], 10))
  }
  return v
}
```
Then in the engine: `JSON.parse(jsonState, reviver)` and `JSON.stringify(state, replacer)`.

For some reason the replacer doesn't get the moment objects but the output of `moment().toJSON()` instead, that's why the regexp.

This solves the problem for me that I automatically get moment objects back to state on load.

**Comment by RanzQ** [comment]
Added PR https://github.com/michaelcontento/redux-storage-engine-localstorage/pull/20

**Comment by michaelcontento** [comment]
closed as #20 has been merged
